### PR TITLE
workflow: Rename Record to Run

### PR DESCRIPTION
**Pull request by andrewwormald**
The Record has morphed and as time has gone on the idea of a workflow "run" and a workflow "Record" (different from WireRecord) have become one in the same. This MR changes the naming so that Record becomes `Run` and `WireRecord` becomes `Record`.

This cleans up the consumer interface / param where you are currently passed a Record but its more like the Run as you can control the run via the RunStateController.

The idea is that this is a breaking change prior to v1.

### CODE REVIEW

File: record.go [hunk 1]
```diff
@@ -1,56 +1,6 @@
 package workflow
 
-import (
-	"context"
-	"testing"
-	"time"
-)
-
-type Record[Type any, Status StatusType] struct {
```
Comment (andrewwormald) [inline record.go:9]: This has all moved to run.go

File: adapters/reflexstreamer/reflex_test.go [hunk 2]
```diff
@@ -9,13 +9,13 @@ import (
...
 	"github.com/luno/reflex/rsql"
-	"github.com/stretchr/testify/require"
-
 	"github.com/luno/workflow"
 	"github.com/luno/workflow/adapters/adaptertest"
 	"github.com/luno/workflow/adapters/memrecordstore"
 	"github.com/luno/workflow/adapters/memrolescheduler"
 	"github.com/luno/workflow/adapters/memtimeoutstore"
+	"github.com/stretchr/testify/require"
+
```
Comment (echarrod) [inline adapters/reflexstreamer/reflex_test.go:18]:
```suggestion
```

Comment (andrewwormald) [reply adapters/reflexstreamer/reflex_test.go:18]: Going to move this import up above the workflow imports

Comment (andrewwormald) [reply adapters/reflexstreamer/reflex_test.go:18]: Ah I see this is because all of these are seen as external mods

**Pull request merged**